### PR TITLE
Safe typecasting

### PR DESCRIPTION
**Pull request by Retler**
### Description
This PR adds safe typecasting. This applies especially to our context handling, where the retrieved value is always a generic `interface{}` type.

### Parent Issue
Issue #108 

### Comments for the Reviewers
- Did I overlook something?
- Is the error handling ok?
- Does it make sense to make bad-path testing for this, and if so, can we do this without having to test every function which retrieves values from context?

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [x] Bugfix
- [ ] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [x] I have updated/added relevant documentation (readme, doc comments, etc).
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/app/app.go [hunk 1]
```diff
@@ -175,7 +177,15 @@ func CheckInsecure(config *Config) {
 // This function is intended to be used for cli operation
 func (app *App) CreateAdminCommand() {
 	ctx := context.Background()
+	// Need to inject requestID manually, as these calls dont pass the ususal middleware
```
Comment (bruno-cc) [inline encryption-service/app/app.go:180]: ``
```suggestion
	// Need to inject requestID manually, as these calls don't pass the usual middleware
```

File: encryption-service/app/permissions_handler.go [hunk 2]
```diff
@@ -58,7 +58,14 @@ func (app *App) GetPermissions(ctx context.Context, request *GetPermissionsReque
 // Grant a user access to an object.
 // The requesting user has to be authorized to access the object.
 func (app *App) AddPermission(ctx context.Context, request *AddPermissionRequest) (*AddPermissionResponse, error) {
-	authStorageTx := ctx.Value(contextkeys.AuthStorageTxCtxKey).(authstorage.AuthStoreTxInterface)
+	authStorageTx, ok := ctx.Value(contextkeys.AuthStorageTxCtxKey).(authstorage.AuthStoreTxInterface)
+	if !ok {
+		err := status.Errorf(codes.Internal, "error encountered while adding permissions")
+		log.Error(ctx, "AddPermission: Could not parse authStorage from context", err)
```
Comment (bruno-cc) [inline encryption-service/app/permissions_handler.go:64]: I think parse is not the right term here and these message is a bit different from the other ones.

**Review by bruno-cc — APPROVED** [review]
See comment about consistent errors message. Otherwise looks fine!

File: encryption-service/app/auth_handlers.go [hunk 3]
```diff
@@ -51,7 +51,14 @@ var methodScopeMap = map[string]authn.ScopeType{
 // lacks the required scope
 func (app *App) AuthenticateUser(ctx context.Context) (context.Context, error) {
 	// Grab method name
-	methodName := ctx.Value(contextkeys.MethodNameCtxKey).(string)
+	methodName, ok := ctx.Value(contextkeys.MethodNameCtxKey).(string)
+	if !ok {
+		err := status.Errorf(codes.Internal, "AuthenticateUser: Internal error during authentication")
+		log.Error(ctx, "Could not parse methodName from context", err)
+
```
[the remark this answers is not in the document]
Comment (Retler) [reply encryption-service/app/auth_handlers.go:58]: fixed

File: encryption-service/app/permissions_handler.go (hunk 2 above)
Comment (Retler) [reply encryption-service/app/permissions_handler.go:64]: I have changed the wording to resemble the other messagtes

File: encryption-service/app/app.go [hunk 4]
```diff
@@ -175,7 +177,15 @@ func CheckInsecure(config *Config) {
...
 	ctx := context.Background()
+	// Need to inject requestID manually, as these calls don't pass the usual middleware
+	requestID, err := uuid.NewV4()
+	if err != nil {
+		log.Fatal(ctx, "Could not generate uuid", err)
+	}
+	ctx = context.WithValue(ctx, contextkeys.RequestIDCtxKey, requestID)
+
 	authStoreTx, err := app.AuthStore.NewTransaction(ctx)
+
```
Comment (bruno-cc) [inline encryption-service/app/app.go:188]: I would remove this space (sorry)

**Review by philip-cc — APPROVED** [review]
Looks great!

**Pull request merged**